src/api/app/models/staging/staging_project.rb

```diff
@@ -133,6 +142,14 @@ def build_state
       :acceptable
     end
 
+    def relevant_checks
+      @relevant_checks ||= checks.where(status_reports: { uuid: repositories.map(&:build_id) })
```

Comment:
why do you assign it to a variable? this is only used once.

`pluck(:build_id)` instead of `map`

You should move it to an [scope](https://guides.rubyonrails.org/active_record_querying.html#scopes)

Reply:
``build_id`` is not a field, it is coming from the backend, you can not pluck it.